daemon/cmd/daemon.go

```diff
@@ -447,7 +447,9 @@ func configureDaemon(ctx context.Context, cleaner *daemonCleanup, params daemonP
 	// restore endpoints before any IPs are allocated to avoid eventual IP
 	// conflicts later on, otherwise any IP conflict will result in the
 	// endpoint not being able to be restored.
-	params.EndpointRestorer.RestoreOldEndpoints()
+	if err := params.EndpointRestorer.RestoreOldEndpoints(); err != nil {
+		return err
+	}
 	bootstrapStats.restore.End(true)
```

Comment:
Let's make sure we also update the bootstrap stats correctly in case of an error:

```suggestion
	err = params.EndpointRestorer.RestoreOldEndpoints();
	bootstrapStats.restore.EndError(err)
	if  err != nil {
		return err
	}
```